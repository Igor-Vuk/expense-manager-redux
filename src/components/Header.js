import React from 'react'
import {NavLink} from 'react-router-dom'

const Header = () => (
  <header>
    <h1>Expense Manager</h1>
    <div>
      <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
    </div> 
    <br/>
    <div>
      <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
    </div>
    <br/>
    <div>
      <NavLink to="/help" activeClassName="is-active">Help</NavLink>
    </div>
    <br/>
  </header>
)

export default Header

