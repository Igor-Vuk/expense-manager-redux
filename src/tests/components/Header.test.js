/* react-test-renderer => it allows us to render our components inside regular javascript code and then we can assert something about what got rendered.  */

import React from 'react'
import Header from '../../components/header'

// React-test-renderer
// --------------------------
// import ReactShallowRenderer from 'react-test-renderer/shallow'
//
// test('should render Header correctly', () => {
//   /* We create a new instance of renderer */
//   const renderer = new ReactShallowRenderer()
//   /* We use render and we define JSX that we wanna render */
//   renderer.render(<Header />)
//   /* We can see rendered output */
//   console.log(renderer.getRenderOutput())
//   /* Jest is gonna create a snapshot of what the rendered header output looked like */
//   expect(renderer.getRenderOutput()).toMatchSnapshot()
// })


// Enzyme
// -------------------------------
/* more complex renderer for react is enzyme. To use it with v16 React we need 'enzyme', 'enzyme-adapter-react-16', 'raf'  which is short for request animation frame. We must also make setupTest and jest.config.json file */


import {shallow} from 'enzyme'

test('should render Header correctly', () => {
  const wrapper = shallow(<Header />)
  expect(wrapper).toMatchSnapshot()
  /* Other examples */
  // expect(wrapper.find('h1').length).toBe(1) 
  // expect(wrapper.find('h1').text()).toBe('Expense Manager') 
})
