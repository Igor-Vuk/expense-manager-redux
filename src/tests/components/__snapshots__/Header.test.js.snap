// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expense Manager
  </h1>
  <div>
    <NavLink
      activeClassName="is-active"
      aria-current="page"
      exact={true}
      to="/"
    >
      Dashboard
    </NavLink>
  </div>
  <br />
  <div>
    <NavLink
      activeClassName="is-active"
      aria-current="page"
      to="/create"
    >
      Create Expense
    </NavLink>
  </div>
  <br />
  <div>
    <NavLink
      activeClassName="is-active"
      aria-current="page"
      to="/help"
    >
      Help
    </NavLink>
  </div>
  <br />
</header>
`;
